include/swaybar/config.h

```diff
@@ -65,6 +76,16 @@ uint32_t parse_position(const char *position);
 char *parse_font(const char *font);
 
 /**
+ * Parse display mode dock|hide|invisible.
+ */
+enum display_mode_types parse_display_mode(const char *display_mode);
+
+/**
+ * Parse hidden state show|hide.
+ */
+enum hidden_states parse_hidden_state(const char *hidden_state);
+
```

Comment:
@calvinkosmatka, suggestion:
```
int parse_enum_string(const char** strings, int n, const char* s);
```
Then put the variable part of the problem into static initialized arrays, and both these function (and future ones) can just call into the one function, catch any error, and cast the return value.

You can use `sizeof(char* mode_enums[])/sizeof(void*)` to have the compiler compute `n`  for you
in individual parse functions when they delegate. The compiler knows this value at compile-time.